src/main/java/com/redhat/devtools/lsp4ij/LSPFileListener.java

```diff
@@ -69,15 +69,22 @@ public void fileClosed(@NotNull FileEditorManager source, @NotNull VirtualFile f
     @Override
     public void propertyChanged(@NotNull VirtualFilePropertyEvent event) {
         if (event.getPropertyName().equals(VirtualFile.PROP_NAME) && event.getOldValue() instanceof String) {
+            String oldValue = (String) event.getOldValue();
+            String newValue = null;
             // A file (Test1.java) has been renamed (to Test2.java) by using Refactor / Rename from IJ
+            // or a "properties" file is changed or saved.
+            if (event.getNewValue() instanceof String) {
+                newValue = (String) event.getNewValue();
+            }
+            if (!oldValue.equals(newValue)) {

```

Comment:
To avoid NPE with oldValue. I will compare old and new value with Objects.equals(oldValue, newValue)